flink-yarn/src/main/java/org/apache/flink/yarn/configuration/YarnConfigOptions.java

```diff
@@ -52,7 +52,74 @@
 		key("yarn.per-job-cluster.include-user-jar")
 			.defaultValue("ORDER");
 
+	/**
+	 * The vcores exposed by YARN.
+	 */
+	public static final ConfigOption<Integer> YARN_VCORES =
+		key("yarn.containers.vcores")
+		.defaultValue(Integer.MAX_VALUE);
```

Comment:
I determine this default by test case, sorry :( 
In fact I want to use `noDefaultValue()` here, but I find this method can be used for `String` only, and there's no `getInteger(ConfigOption<Integer> option, int overrideDefault)` in `Configuration`.

I think it may be better to add `public ConfigOption<T> noDefaultValue()` in `ConfigOptions` and add `public int getString(ConfigOption<Integer> configOption, int overrideDefault)` in `Configuration` . What do you think? THX :)